pkg/platform/kube/controller/nucliofunction.go

```diff
@@ -394,14 +394,24 @@ func (fo *functionOperator) setFunctionError(ctx context.Context,
 		"functionName", function.Name,
 		"err", err)
 
-	if setStatusErr := fo.setFunctionStatus(detachedContext, function, &functionconfig.Status{
+	functionStatus := &functionconfig.Status{
 		Logs:                   function.Status.Logs,
 		State:                  functionErrorState,
 		Message:                errors.GetErrorStackString(err, 10),
 		InternalInvocationURLs: []string{},
 		ExternalInvocationURLs: []string{},
 		EnrichedNodeSelector:   function.Status.EnrichedNodeSelector,
-	}); setStatusErr != nil {
+	}
+
+	// if function is unhealthy, it might be invokable, and
+	// it can become ready, so enrich invocation status
+	if functionErrorState == functionconfig.FunctionStateUnhealthy && resources != nil {
+		if err := fo.populateFunctionInvocationStatus(function, functionStatus, resources); err != nil {
+			return errors.Wrap(err, "Failed to populate function invocation status")
+		}
+	}
```

Comment:
There's only 1 place the function state can be set to unhealthy (line 275). Why not handle it specifically there instead of in the more general `setFunctionError` function?

Reply:
@TomerShor because status object is defined there. I think, if doing it properly, the whole code of this function should be refactored and creation of any additional status object should be avoided because status is already a part of the function. This way to fix it is the least intrusive I can think of.